rfc/mediachain-rfc-2.md

```diff
@@ -0,0 +1,562 @@
+network to fork the blockchain into two or more competing chains.
+The partition healing protocol must merge these competing blockchains
+once connectivity is reestablished.
+
+The protocol can proceed by choosing the longest chain, breaking ties
+with a deterministic scheme so that all nodes can make the same
+choices independently.  Once choosing the new primary blockchain, the
+orphaned chains can be merged by replaying necessary transactions and
+rerunning the block commit protocol.
+
+### Byzantine Failures
+
+#### Bug or Malice?
+
+Invariably, the system will experience bugs or malicious behavior with
+arbitrary failure characteristics. We consider cases where the
+adversarial process (be it bug, murphy or a threat actor) controls a
+number of nodes in the system.
+
+In general, the adversarial process is observable to the system
+through message exchange. An adversarial action can be
+a message injection, a broadcast message suppression or an arbitrary
+long combination of the two.
+
+#### Message Suppression Attacks
+
+The peer-to-peer broadcast of transactions in the core network routes
+through multiple paths. As such, it can withstand a number of
+nodes trying to deny or disrupt the service by suppressing
+broadcast messages. 
+
+At worst, clients of adversarial nodes will be denied service.
+Affected clients can easily recover by connecting to another Peer for
+service. If too many nodes are dropping transactions so as to affect
+the system as a whole, the disruption would be big enough for a metric
+event visible to system monitoring. At which point, intervention
+by tier-1 node operators can restore the problem,.
+
+#### Message Injection Attacks
+
+At a base level, protocol implementation can reject injection of all
+obviously invalid messages that do not conform to rpc signatures.
+Nodes injecting such messages can be immediately disconnected and be
+considered failed. In the following we are concerned with the effects
+of well-formed messages that can affect protocol behavior at the
+algorithmic level.
+
+We can further characterize such injections as poisoning
+and flooding attacks. In a poisoning attack, the adversarial
+process attempts to inject messages that can corrupt the state
+of the Mediachain. In a flooding attack, the adversarial
+process attempts to flood the network with messages that
+hijack blockchain algorithms and disrupt service.
+
+#### Client Injection Attacks
+
+The network is open to all clients, thus we are more concerned
+about malicious behavior than genuine bugs in client activity.
+
+A client can attempt a poisoning attack by injecting invalid
+transactions. Such transactions may refer to records not written
+to the datastore or be duplicates of older transactions, etc.
+The system can be rendered robust against transaction poisoning
+attacks by means of transaction verification. All peers verify
+client transactions before broadcasting them to the core network.
```

Comment:
Can you clarify this? Do you mean the Peer verification process under `Extending the Blockchain`? I like consistently using capitalized `Peer` to mean tier-1 nodes